src/web/handlers.rs

```diff
@@ -402,3 +402,36 @@ pub fn put_bso(bso_req: BsoPutRequest) -> impl Future<Item = HttpResponse, Error
 pub fn get_configuration(creq: ConfigRequest) -> impl Future<Item = HttpResponse, Error = Error> {
     future::result(Ok(HttpResponse::Ok().json(creq.limits)))
 }
+
+/** Returns a status message indicating the state of the current server
+ *
+ */
+pub fn heartbeat(hb: HeartbeatRequest) -> impl Future<Item = HttpResponse, Error = Error> {
+    let mut checklist = HashMap::new();
+    checklist.insert(
+        "version".to_owned(),
+        Value::String(env!("CARGO_PKG_VERSION").to_owned()),
+    );
+    hb.db.check().then(|response| match response {
+        Ok(result) => {
+            if result {
+                checklist.insert("database".to_owned(), Value::from("Ok"));
+            } else {
+                checklist.insert("database".to_owned(), Value::from("Err"));
+                checklist.insert(
+                    "database_msg".to_owned(),
+                    Value::from("check failed without error"),
+                );
+            };
+            let status = if result { "Ok" } else { "Err" };
+            checklist.insert("status".to_owned(), Value::from(status));
+            HttpResponse::Ok().json(checklist)
+        }
+        Err(e) => {
+            checklist.insert("status".to_owned(), Value::from("Err"));
+            checklist.insert("database".to_owned(), Value::from("Unknown"));
+            checklist.insert("status_msg".to_owned(), Value::from(e.to_string()));
```

Comment:
In megaphone I originally included the error like this in the json, then opted to remove it from there just in case the db error required sanitizing. A case like this is a good example, exposing the GCP project/instance info: https://sentry.prod.mozaws.net/operations/syncstorage-stage/issues/6625678/?query=session

Instead it's just logged